### PR TITLE
DEV: Remove the upload processing placeholder

### PR DESCRIPTION
**Pull request by CvX**
Just use a the single "Uploading…" placeholder. End user doesn't benefit much from knowing at what point in the processing pipeline their upload is. 😃

### CODE REVIEW

**Review by martin-brennan — CHANGES_REQUESTED** [review]
Hmm while I agree it's not really necessary to change the placeholder every time a different preprocessor is done, IMO we should still show Processing... until all of the preprocessors are done. Then we have parity with the status at the bottom of the composer that says "Processing Upload" too.

File: app/assets/javascripts/discourse/app/mixins/composer-upload-uppy.js [hunk 1]
```diff
@@ -99,7 +99,7 @@ export default Mixin.create(ExtendableUploader, UppyS3Multipart, {
 
   _bindUploadTarget() {
     this.set("inProgressUploads", []);
-    this.placeholders = {};
+    this.placeholders = new Map();
```
Comment (martin-brennan) [inline app/assets/javascripts/discourse/app/mixins/composer-upload-uppy.js:102]: Nice, map works much better here 👍

File: app/assets/javascripts/discourse/app/mixins/composer-upload-uppy.js [hunk 2]
```diff
@@ -447,42 +448,8 @@ export default Mixin.create(ExtendableUploader, UppyS3Multipart, {
         );
       });
 
-    this._onPreProcessProgress((file) => {
```
Comment (martin-brennan) [inline app/assets/javascripts/discourse/app/mixins/composer-upload-uppy.js:450]: I think we could still leave this off, and just change this code:

https://github.com/discourse/discourse/blob/bb342bafe9a9d41f912c49e2eeda3edb66c60b11/app/assets/javascripts/discourse/app/mixins/composer-upload-uppy.js#L304-L309

To use the `processingPlaceholder` from the beginning...then when `_onPreProcessComplete` is fired replace it with the uploading placeholder

File: app/assets/javascripts/discourse/app/mixins/composer-upload-uppy.js [hunk 3]
```diff
@@ -447,42 +448,8 @@ export default Mixin.create(ExtendableUploader, UppyS3Multipart, {
...
-        placeholderData.uploadPlaceholder,
-        placeholderData.processingPlaceholder
-      );
-
-      // Safari applies user-defined replacements to text inserted programmatically.
-      // One of the most common replacements is ... -> …, so we take care of the case
-      // where that transformation has been applied to the original placeholder
-      this.appEvents.trigger(
-        `${this.composerEventPrefix}:replace-text`,
-        placeholderData.uploadPlaceholder.replace("...", "…"),
```
Comment (martin-brennan) [inline app/assets/javascripts/discourse/app/mixins/composer-upload-uppy.js:470]: I think this `replace()` could move to inside the `_uploadPlaceholder` function instead.

**Comment by CvX** [comment]
> IMO we should still show Processing... until all of the preprocessors are done. Then we have parity with the status at the bottom of the composer that says "Processing Upload" too.

I'd change that composer status to display only "Uploading" too 😅 Does it make any difference to the user if it's processing vs uploading-uploading? Do other platforms differentiate between under-the-hood steps? 😃

**Comment by martin-brennan** [comment]
@CvX Yes actually, Slack does for example:

image


Discord as well:

image

Google Drive:

image


So yes, we should keep the differentiation between the two phases; especially since at some point we may want to add post-processing to the pipeline. Happy to have an internal design/product topic about it too if you'd like.

**Pull request closed without merging**